### PR TITLE
Metric filters for distinct values queries

### PR DESCRIPTION
**Pull request by courtneyholcomb**
Completes SL-1939

### Description
Enable metric filters for distinct values queries (queries without metrics).

### CODE REVIEW

**Comment by tlento** [comment]
Trino is once again unhappy about something. Another non-ANSI but broadly-supported group by expression, perhaps?

**Comment by courtneyholcomb** [comment]
> Trino is once again unhappy about something. Another non-ANSI but broadly-supported group by expression, perhaps?

@tlento DANGIT. Updated!

**Comment by tlento** [comment]
Argh, this fell off my radar, sorry.

Is this commit set correct? It looks like you need a rebase, I'm seeing things that look like duplicates of earlier work. If so, you can just point me to the first thing I should read and I can go from there while the PR itself gets fixed up.

**Comment by courtneyholcomb** [comment]
> Argh, this fell off my radar, sorry.
> 
> Is this commit set correct? It looks like you need a rebase, I'm seeing things that look like duplicates of earlier work. If so, you can just point me to the first thing I should read and I can go from there while the PR itself gets fixed up.

Dangit you're right! The first commit to look at is [Add METRIC property to all_properties](https://github.com/dbt-labs/metricflow/pull/1107/commits/3a2156e03b5728d40203350a3d60898b575adacb). Thank you!!

File: tests/integration/test_cases/itest_dimensions.yaml [hunk 1]
```diff
@@ -332,4 +332,30 @@ integration_test:
       GROUP BY
         {{ render_date_trunc("v.ds", TimeGranularity.MONTH) }}
     )
+---
+integration_test:
+  name: distinct_values_query_with_metric_filter
+  description: Query withought metrics using a metric filter
```
Comment (tlento) [inline tests/integration/test_cases/itest_dimensions.yaml:338]: micro-nit: without

File: metricflow/dataflow/builder/dataflow_plan_builder.py [hunk 2]
```diff
@@ -640,9 +640,7 @@ def _build_plan_for_distinct_values(self, query_spec: MetricFlowQuerySpec) -> Da
             )
 
             query_level_filter_specs = filter_spec_factory.create_from_where_filter_intersection(
-                filter_location=WhereFilterLocation.for_query(
-                    tuple(metric_spec.reference for metric_spec in query_spec.metric_specs)
-                ),
+                filter_location=WhereFilterLocation.for_query(metric_references=tuple()),
```
Comment (tlento) [inline metricflow/dataflow/builder/dataflow_plan_builder.py:643]: You know, it took me way too long to figure out why there were no metric specs in this query. 🤦

**Review by tlento — APPROVED** [review]
Forgotten no more! Sorry about that, every day I was like "oh I'll do this before I stop today" and every day I filled my brain with dataflow plan builder implementation details and forgot all about everything else, so I'm going back to the review first, code later order of operations.

File: tests/integration/test_cases/itest_dimensions.yaml (hunk 1 above)
Comment (courtneyholcomb) [reply tests/integration/test_cases/itest_dimensions.yaml:338]: Idk I kinda like my creative spelling.

**Pull request merged**